pkg/kv/kvclient/kvcoord/txn_interceptor_span_refresher.go

```diff
@@ -447,28 +448,30 @@ func (sr *txnSpanRefresher) maybeRefreshPreemptivelyLocked(
 
 	canRefreshTxn, refreshTxn := roachpb.PrepareTransactionForRefresh(ba.Txn, ba.Txn.WriteTimestamp)
 	if !canRefreshTxn || !sr.canAutoRetry {
-		return roachpb.BatchRequest{}, newRetryErrorOnFailedPreemptiveRefresh(ba.Txn)
+		return roachpb.BatchRequest{}, newRetryErrorOnFailedPreemptiveRefresh(ba.Txn, sr.refreshFootprint.asSlice())
```

Comment:
I'm not confident that `asSlice` returns immutable memory. For example, look here:

https://github.com/cockroachdb/cockroach/blob/66903db87e4712a28bd2dc07424cfafecd91ae23/pkg/kv/kvclient/kvcoord/condensable_span_set.go#L50-L57

This seems to mutate the memory backing `s.s` in-place, so by referencing it from the error here and below, aren't we causing a data race?